tools/src/bin/dependency_check.rs

```diff
@@ -0,0 +1,290 @@
+            }
+        } else if file.file_type()?.is_file() {
+            let path = file.path();
+            let filename = path
+                .file_name()
+                .unwrap_or_default()
+                .to_str()
+                .unwrap_or_default();
+            if re.is_match(filename) {
+                libraries.push(path);
+            }
+        } else {
+            unreachable!();
+        }
+    }
+    Ok(libraries)
+}
+
+fn download_ffmpeg(executable_dir: &Path, ffmpeg_url: &str) -> Result<()> {
+    let response = get(ffmpeg_url)
+        .with_context(|| format!("Failed to download FFmpeg libraries. URL: {ffmpeg_url}"))?;
+    let content = response.bytes()?;
+
+    fs::write(executable_dir.join(FFMPEG_ARCHIVE_NAME), &content)
+        .with_context(|| "Failed to save downloaded FFmpeg libraries to file")?;
+    Ok(())
+}
+
+#[cfg(target_os = "linux")]
+fn check_ffmpeg(required_ffmpeg_version: &str) -> Result<bool> {
+    check_ffmpeg_command(required_ffmpeg_version)
+}
+
+#[cfg(target_os = "macos")]
+fn check_ffmpeg(required_ffmpeg_version: &str) -> Result<bool> {
+    let command_result = check_ffmpeg_command(required_ffmpeg_version)?;
+    if !command_result {
+        info!("Checking if ffmpeg is installed as homebrew keg-only");
+        return check_ffmpeg_homebrew(required_ffmpeg_version);
+    }
+    Ok(command_result)
+}
+
+fn check_ffmpeg_command(required_ffmpeg_version: &str) -> Result<bool> {
+    let ffmpeg_result = Command::new("ffmpeg").arg("-version").output();
+    match ffmpeg_result {
+        Ok(ffmpeg_output) => {
+            let ffmpeg_output = String::from_utf8(ffmpeg_output.stdout)?.trim().to_string();
+            match match_ffmpeg_version(&ffmpeg_output) {
+                Some(version) if version == required_ffmpeg_version => Ok(true),
+                Some(version) => {
+                    warn!(
+                        installed_ffmpeg_version = version,
+                        required_ffmpeg_version,
+                        "Inatelled version doesn't match the required version."
+                    );
+                    Ok(false)
+                }
+                None => {
+                    warn!("Failed to parse FFmpeg version.");
+                    Ok(false)
+                }
+            }
+        }
+        Err(_) => {
+            warn!("Failed to run FFmpeg.");
+            Ok(false)
+        }
+    }
+}
+
+fn match_ffmpeg_version(ffmpeg_output: &str) -> Option<&str> {
+    let re = Regex::new(r"(?m)^ffmpeg version \D*(\d+\.\d+)")
+        .expect("Failed to compile regular expression");
+    let caps = re.captures(ffmpeg_output);
+    match caps {
+        Some(caps) => caps.get(1).map(|cap| cap.as_str()),
+        None => {
+            warn!("Failed to parse FFmpeg version.");
+            None
+        }
+    }
+}
+
+#[cfg(target_os = "macos")]
+fn check_ffmpeg_homebrew(required_ffmpeg_version: &str) -> Result<bool> {
+    let required_ffmpeg_version_brew =
```

Comment:
When looking at this code, I don't know what `required_ffmpeg_version`  is what it will be after removing before the dot